sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_base.py

```diff
@@ -0,0 +1,347 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from __future__ import annotations
+from typing import Tuple, Union, TYPE_CHECKING, Dict
+from abc import ABC, abstractmethod
+
+if TYPE_CHECKING:
+    try:
+        from uamqp import types as uamqp_types
+    except ImportError:
+        uamqp_types = None
+
+class AmqpTransport(ABC):   # pylint: disable=too-many-public-methods
+    """
+    Abstract class that defines a set of common methods needed by producer and consumer.
```

Comment:
pretty much yes. I added a few more methods and don't need to use a few of them. So going to remove unnecessary methods + add correct ones here + updating docstring/types to use SB instead of EH